ddsp/training/docker/user_interaction.py

```diff
@@ -0,0 +1,150 @@
+
+  config_path = input('\nInsert the path to a configuration file '
+                      'or skip for the default one:\n')
+  if config_path == '':
+    config_path = './config_single_vm.yaml'
+  input_dict['config_path'] = config_path
+
+  image_uri = input('\nInsert an IMAGE URI. '
+                    'The template is: gcr.io/<GCP_PROJECT_ID>'
+                    '/<IMAGE_REPO_NAME>:<IMAGE_TAG> :\n')
+  input_dict['image_uri'] = image_uri
+
+  job_name = input('\nInsert a job name:\n')
+  input_dict['job_name'] = job_name
+
+  region = input('\nInsert the region you want to train '
+                  'your model in or skip for the default '
+                  'value (europe-west4): ')
+  if region == '':
+    region = 'europe-west4'
+  input_dict['region'] = region
+
+  batch_size = input('\nInsert batch size '
+                      'or skip for the default value (128): ')
+  if batch_size == '':
+    batch_size = '128'
+  input_dict['batch_size'] = batch_size
+
+  learning_rate = input('\nInsert the learning rate '
+                        'or skip for the default value (0.001): ')
+  if learning_rate == '':
+    learning_rate = '0.001'
+  input_dict['learning_rate'] = learning_rate
+
+  no_of_steps = input('\nInsert the number of steps for training '
+                      'or skip for the default value (15000): ')
+  if no_of_steps == '':
+    no_of_steps = '15000'
+  input_dict['no_of_steps'] = no_of_steps
+
+  steps_per_save = input('\nInsert the number of steps per save '
+                          'or skip for the default value (300): ')
+  if steps_per_save == '':
+    steps_per_save = '300'
+  input_dict['steps_per_save'] = steps_per_save
+
+  steps_per_summary = input('\nInsert the number of steps per summary '
+                            'or skip for the default value (300): ')
+  if steps_per_summary == '':
+    steps_per_summary = '300'
+  input_dict['steps_per_summary'] = steps_per_summary
+
+  early_stop_loss_value = input('\nInsert the early stop loss value '
+                                'or skip for the default value (5): ')
+  if early_stop_loss_value == '':
+    early_stop_loss_value = '5'
+  input_dict['early_stop_loss_value'] = early_stop_loss_value
+
+  return input_dict
+
+def build_image(input_dict):
+  """Builds the docker image"""
+  build_command = 'docker build -f Dockerfile -t '\
+                  + input_dict['image_uri'] + ' ./'
+  os.system(build_command)
+
+def push_image(input_dict):
+  """Pushes the docker image on Google Cloud Registry"""
+  pushing_image = 'docker push ' + input_dict['image_uri']
+  os.system(pushing_image)
+
+def submit_job(input_dict):
+  """Submits the job on AI Platform"""
+  os.system('export PATH=/usr/local/google/home/$USER/.local/bin:$PATH')
+
+  submitting_job = 'gcloud beta ai-platform jobs submit training '\
+  + input_dict['job_name'] + ' --region ' + input_dict['region']\
+  + ' --master-image-uri ' + input_dict['image_uri']\
```

Comment:
for all of these input_dict prompts, it would be nice if you have this script also have the option of taking in the flags, so it just prompts for flags that have not been specified.

that way this `ddsp_ai_platform` script that's installed with the pip package can be the main entry point for launching jobs on ai-platform, and can be automated with flags if people don't want to manually input everything.

Reply:
Done